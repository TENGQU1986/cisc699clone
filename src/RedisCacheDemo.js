import React from 'react'
import withoutCache from './images/withoutCache.png'
import withCache from './images/withCache.png'
import './App.css'

const RedisCacheDemo = () => {
    return (
        <div className="container">
            <h1 className="cache-header">Redis Cache Demo</h1>
            <h3>You could download the source code from <a href="https://github.com/TENGQU1986/redis-cache-demo">here</a>.</h3>
            <br></br>
            <h4>First, let's see if we just simply fetch data every single time when page initial rendering. It takes pretty much the same amount of time to fetch data, in this particular case, around 800ms to 3 seconds. By the way, this API is a fake API provided by Jsonplaceholder and it contains over 35,000 items.</h4>
            <img src={withoutCache} alt="" width="1000" height="900"/>
            <h4>Then we just fetch the data through API at the very first time and cache those data in the redis cache database, then you will see the fetch time is extremely faster than fetching from the API</h4>
            <img src={withCache} alt="" width="1000" height="900"/>


            

        </div>
    )
}

export default RedisCacheDemo
