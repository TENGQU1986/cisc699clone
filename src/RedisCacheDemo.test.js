import React from 'react';
import { render, fireEvent, act } from "@testing-library/react";
import RedisCacheDemo from "./Comments";

describe('rendering correctly', () => {

    it('should render correctly', () => {
        const { container } = render(<RedisCacheDemo />);
        expect(container).toMatchSnapshot();
    })
    
})