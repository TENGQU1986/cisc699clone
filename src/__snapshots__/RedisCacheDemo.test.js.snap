// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering correctly should render correctly 1`] = `
<div>
  <div
    class="comments"
  >
    <span
      class="comment_summary"
    >
      Comments: 
      2
       
    </span>
    <div
      class="input_area"
    >
      <form>
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          style="margin: 8px;"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
            data-shrink="true"
            for="standard-full-width"
            id="standard-full-width-label"
          >
            Comment
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-describedby="standard-full-width-helper-text"
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              id="standard-full-width"
              placeholder="Type comment here"
              type="text"
              value=""
            />
          </div>
          <p
            class="MuiFormHelperText-root"
            id="standard-full-width-helper-text"
          >
            Don't contain personal information
          </p>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            Post your Comment
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </form>
    </div>
    <div
      class="comment"
    >
      <div>
        <div
          class="comment_card"
        >
          <span
            class="comment_author"
          >
            terryqu2021@gmail.com
          </span>
          <h4>
            Hello there!
          </h4>
        </div>
      </div>
      <div>
        <div
          class="comment_card"
        >
          <span
            class="comment_author"
          >
            quteng.bo@gmail.com
          </span>
          <h4>
            thanks for sharing
          </h4>
        </div>
      </div>
    </div>
  </div>
</div>
`;
